normits_demand/models/forecasting/EDGE_growth/loading.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""
+Created on: 26/05/2023
+Updated on:
+
+Original author: Ben Taylor
+Last update made by:
+Other updates made by:
+
+File purpose:
+
+"""
+# Built-Ins
+import dataclasses
+import pickle
+from pathlib import Path
+# Third Party
+import pandas as pd
+
+# Local Imports
+# pylint: disable=import-error,wrong-import-position
+from normits_demand.models.forecasting import forecast_cnfg
+from normits_demand.utils import file_ops
+import ticket_splits
+
+
+# pylint: enable=import-error,wrong-import-position
+
+# # # CONSTANTS # # #
+
+# # # CLASSES # # #
+
+
+@dataclasses.dataclass
+class GLobalVars:
+    demand_segments: pd.DataFrame
+    purposes: list
+    norms_segments: list
+    all_segments: list
+    ticket_type_splits: pd.DataFrame
+    station_tlcs: pd.DataFrame
+    time_periods: tuple = ("AM", "IP", "PM", "OP")
+
+    def keys(self):
+        return [field.name for field in dataclasses.fields(self)]
+
+
+# # # FUNCTIONS # # #
+def load_globals(params: forecast_cnfg.EDGEParameters) -> GLobalVars:
+    """
+    Load in global variables for the process (i.e. variables which don't vary by year/time_period/segment).
+    """
+    demand_segments = file_ops.read_df(params.demand_segments)
+    norms_segments = (
+        demand_segments.loc[demand_segments["ModelSegment"] == 1][
+            ["Segment"]
+        ]
+        .drop_duplicates()
+        .values.tolist()
+    )
+    purposes = demand_segments["Purpose"].drop_duplicates().to_list()
+    norms_segments = [
+        segment for sublist in norms_segments for segment in sublist
+    ]
+    # all segments
+    all_segments = demand_segments["Segment"].to_list()
+    demand_segments.loc[:, "ToHome"] = demand_segments["ToHome"].astype(
```

Comment:
I started defaulting to .loc to avoid all of the error messages on index vs slicer or whatever it comes up with. Is there a fundamental issue with using .loc or is it just a readbility thing?